zerver/tests/test_scheduled_messages.py

```diff
@@ -241,6 +242,14 @@ def test_too_late_to_deliver_scheduled_message(self) -> None:
                 scheduled_message, logger, expected_failure_message
             )
 
+            # Verify that the user was sent a message informing them about
+            # the failed scheduled message.
+            realm = scheduled_message.sender.realm
+            msg = most_recent_message(scheduled_message.sender)
+            self.assertEqual(msg.recipient.type, msg.recipient.PERSONAL)
+            self.assertEqual(msg.sender_id, self.notification_bot(realm).id)
+            self.assertIn("Scheduled send time has already passed", msg.content)
```

Comment:
Yeah I'm honestly not sure what's best here. I'm 50/50 on whether I think the UI for this will just display a "failed" icon without actually displaying the message.

The actual explanation for this error is of the form "This Zulip Server either got its queue backlogged or otherwise was experiencing a (partial) outage of the send later service when this message was supposed to be sent, and we decided to cancel the message and notify you rather than sending it more than 15 minutes late". I don't know how to explain this to a user concisely.

Reply:
Updated this for @alya's suggestion [here on CZO](https://chat.zulip.org/#narrow/stream/101-design/topic/scheduled.20message.20error.20handling/near/1568030).